pubsub/pulsar/pulsar.go

```diff
@@ -101,16 +108,23 @@ func (p *Pulsar) Init(metadata pubsub.Metadata) error {
 	if err != nil {
 		return err
 	}
-	client, err := pulsar.NewClient(pulsar.ClientOptions{
-		URL:                        fmt.Sprintf("pulsar://%s", m.Host),
+	var pulsarURL string = m.Host
+	if !strings.HasPrefix(m.Host, "http") {
```

Comment:
Use "http://" to avoid ambiguity when the host contains http in the name  (unlikely, but possible). Also, what about https?

Reply:
So what if I check both `http:// `and `https://` for `m.Host`?